--- a/hip_fp8.h
+++ b/hip_fp8.h
@@ -82,7 +82,13 @@
     return *reinterpret_cast<const hip_fp8_e4_t *>(&data);
   }
   __device__ operator float() const {
-    return static_cast<float>(static_cast<hip_fp8_e4_t>(*this));
+    constexpr __hip_fp8_interpretation_t interp =
+#if (TILELANG_FP8_E4M3_VARIANT == TILELANG_FP8_E4M3_VARIANT_FNUZ)
+        __HIP_E4M3_FNUZ;
+#else
+        __HIP_E4M3;
+#endif
+    return __half2float(__half(__hip_cvt_fp8_to_halfraw(data, interp)));
   }
 };
 
@@ -111,7 +117,13 @@
     return *reinterpret_cast<const hip_fp8_e5_t *>(&data);
   }
   __device__ operator float() const {
-    return static_cast<float>(static_cast<hip_fp8_e5_t>(*this));
+    constexpr __hip_fp8_interpretation_t interp =
+#if (TILELANG_FP8_E5M2_VARIANT == TILELANG_FP8_E5M2_VARIANT_FNUZ)
+        __HIP_E5M2_FNUZ;
+#else
+        __HIP_E5M2;
+#endif
+    return __half2float(__half(__hip_cvt_fp8_to_halfraw(data, interp)));
   }
 };
 // Note: E8M0 types are not supported in current HIP version
